pydantic/generics.py

```diff
@@ -99,15 +107,48 @@ def __concrete_name__(cls: Type[Any], params: Tuple[Type[Any], ...]) -> str:
         """
         This method can be overridden to achieve a custom naming scheme for GenericModels
         """
-        param_names = [param.__name__ if hasattr(param, '__name__') else str(param) for param in params]
+        try:
```

Comment:
is there a reason this needs to be in this function? Couldn't it be moved out to module level.

Reply:
I used `pydantic.typing.display_as_type` and added a case to that instead, which cleans this whole function up quite a lot.  Hope it's clearer now.